### PR TITLE
Adds Azure cloud provider config

### PR DESCRIPTION
**Pull request by r7vme**
this config used by Kubernetes Azure cloud provider.

### CODE REVIEW

File: service/key/key.go [hunk 1]
```diff
@@ -22,13 +38,48 @@ func KeyVaultName(customObject azuretpr.CustomObject) string {
 	return customObject.Spec.Azure.KeyVault.Name
 }
 
+// MasterSecurityGroupName returns name of the security group attached to master subnet.
+func MasterSecurityGroupName(customObject azuretpr.CustomObject) string {
```
Comment (kopiczko) [inline service/key/key.go]: Let's track adding those to ARM template parameters as well.

File: service/cloudconfig/cloud_config.go [hunk 2]
```diff
@@ -48,6 +54,53 @@ func (me *MasterExtension) VerbatimSections() []k8scloudconfig.VerbatimSection {
 	return nil
 }
 
+// masterCloudProviderConf returns Kubernetes cloud provider config for Azure.
+func (me *MasterExtension) masterCloudProviderConf() (k8scloudconfig.FileAsset, error) {
```
Comment (kopiczko) [inline service/cloudconfig/cloud_config.go]: Please add TODO we should make it `cloudProviderConf` as soon as we start creating masters and worker in the same subnet and SG.

Comment (r7vme) [reply service/cloudconfig/cloud_config.go]: Hmm, do we plan to use same subnet?

Comment (kopiczko) [reply service/cloudconfig/cloud_config.go]: Did I misunderstand you? I thought you suggested merging those two guys:


* pawel-cluster-VirtualNetwork-MasterSubnet
* pawel-cluster-VirtualNetwork-WorkerSubnet

Comment (r7vme) [reply service/cloudconfig/cloud_config.go]: i forgot why i suggested this... if only for simplicity, Than we can avoid this and leave as it is right now as two networks will allow us properly assign security rules for master.

I mean for master we need to open some extra ports so proper is to have it done only for master subnet. Sorry for confusion :)

File: service/key/key.go (hunk 1 above)
Comment (r7vme) [reply service/key/key.go]: Can we add this in TODO?

I can do this as separate PR, because it's quite big change.

Comment (kopiczko) [reply service/key/key.go]: Yes, TODO is ok.

**Pull request merged**